src/plugins/discover/public/application/main/components/sidebar/discover_sidebar.tsx

```diff
@@ -196,16 +196,20 @@ export function DiscoverSidebarComponent({
     }
   }, [paginate, scrollContainer, unpopularFields]);
 
-  const fieldTypes = useMemo(() => {
+  const { fieldTypes, presentFieldTypes } = useMemo(() => {
     const result = ['any'];
+    const dataViewFieldTypes = new Set<string>();
     if (Array.isArray(fields)) {
       for (const field of fields) {
-        if (result.indexOf(field.type) === -1) {
-          result.push(field.type);
+        if (field.type !== '_source') {
+          dataViewFieldTypes.add(field.type);
```

Comment:
This is complicated and we should simplify later on. 

So Es typed fields `text` and `keyword` are mapped to the Kibana field type string. So we need to split up this type. I've added code here so it should work, but feel free to extract the extraction of the actual used type to a separate function, so it's better readable code. Apart from that it's now really nice and functional 👍 
```suggestion
          // If it's a string type, we want to distinguish between keyword and text
          // For this purpose we need the ES type
          const type =
            field.type === 'string' &&
            field.esTypes &&
            ['keyword', 'text'].includes(field.esTypes[0])
              ? field.esTypes?.[0]
              : field.type;
          // _id and _index would map to string, that's why we don't add the string type here
          if (type && type !== 'string') {
            dataViewFieldTypes.add(type);
          }
```